### PR TITLE
Refactor/issue 656 dbreeze part 2

### PR DESCRIPTION
**Pull request by Aprogiena**
Part 2 of DBreeze refactoring introduces new way of working with DBreeze to block and index repositories.

Depends on PR #668

### CODE REVIEW

**Comment by dangershony** [comment]
pending conflicts

File: Stratis.Bitcoin.Features.BlockStore/BlockRepository.cs [hunk 1]
```diff
@@ -71,40 +67,39 @@ public virtual BlockStoreRepositoryPerformanceCounter PerformanceCounterFactory(
...
+        public virtual Task InitializeAsync()
         {
             this.logger.LogTrace("()");
-            var genesis = this.network.GetGenesis();
+            Block genesis = this.network.GetGenesis();
 
-            var sync = this.session.Execute(() =>
+            Task task = Task.Run(() =>
             {
-                this.session.Transaction.SynchronizeTables(this.tableNames.ToList());
```
Comment (dangershony) [inline Stratis.Bitcoin.Features.BlockStore/BlockRepository.cs]: no need for `SynchronizeTables`?

File: Stratis.Bitcoin.Features.BlockStore/BlockStoreLoop.cs [hunk 2]
```diff
@@ -147,7 +147,7 @@ public class BlockStoreLoop
                     throw new BlockStoreException($"You need to rebuild the {this.StoreName} database using -reindex-chainstate to change -txindex");
 
                 if (this.storeSettings.TxIndex)
-                    await this.BlockRepository.SetTxIndex(this.storeSettings.TxIndex);
+                    await this.BlockRepository.SetTxIndexAsync(this.storeSettings.TxIndex);
```
Comment (dangershony) [inline Stratis.Bitcoin.Features.BlockStore/BlockStoreLoop.cs]: Maybe add `.ConfigureAwait(false)`

File: Stratis.Bitcoin.Features.BlockStore/BlockRepository.cs (hunk 1 above)
Comment (Aprogiena) [reply Stratis.Bitcoin.Features.BlockStore/BlockRepository.cs]: before there was one transaction,
ST is per transaction
so now you do ST on per transaction basis and you have to do it
just not like before, which was global

**Pull request merged**